src/ASM/LR/ASMu2Dmx.C

```diff
@@ -953,7 +954,13 @@ bool ASMu2Dmx::evalSolution (Matrix& sField, const IntegrandBase& integrand,
 
     // Now evaluate the solution field
     Vector solPt;
-    if (!integrand.evalSol(solPt,fe,Xnod*fe.basis(geoBasis),
+
+    // Cartesian coordinates of current integration point
+    fe.u = gpar[0][i];
+    fe.v = gpar[1][i];
+    utl::Point X4(Xnod*fe.basis(geoBasis),{fe.u,fe.v});
+
+    if (!integrand.evalSol(solPt,fe,X4,
```

Comment:
Similar fix should be done for other ASM-classes also then, for consistency

Reply:
Already there for tensor ASMs, added for all LR ASMs.